contracts/CurrencyNetwork.sol

```diff
@@ -558,6 +554,24 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
         isNetworkFrozen = true;
     }
 
+    function supportsInterface(
```

Comment:
I would say this should be in `CurrencyNetworkInterface`

Reply:
Interfaces *should* (and since I changed `CurrencyNetworkInterface` to be an actual interface also *can*) not contain implementation or inherit from other interfaces or contracts.

What we *could* do is provide a "base" CurrencyNetwork and implement all common/generic logic there.
The CurrencyNetwork Contract has around 1.500 lines currently and is a good candidate for refactoring in any case.

Not really sure if the time invest is justified though.

Reply:
"What we could do is provide a "base" CurrencyNetwork and implement all common/generic logic there."

I had that in mind, seeing the `CurrencyNetworkInterface` was not an actual `interface` but it is fine if you don't, and it could be part of a different task.